protocols/request-response/src/handler/protocol.rs

```diff
@@ -111,6 +111,7 @@ where
                 if let Ok(response) = self.response_receiver.await {
                     let write = self.codec.write_response(&protocol, &mut io, response);
                     write.await?;
+                    io.close().await?;
                 }
             }
```

Comment:
Is there a reason why this one isn't just at the end of the upgrade? Then it would be called under all circumstances, even if we fail to provide a response because the sender has been dropped.

```suggestion
                }
            }            
            io.close().await?;
```

Reply:
If no response is sent, there is no possibility for problems like https://github.com/libp2p/rust-libp2p/pull/1606 to occur. Of course, there is no harm in explicitly closing the substream anyway.